examples/hellogrpc/BUILD

```diff
@@ -41,28 +40,6 @@ k8s_service(
     template = "service.yaml",
 )
 
-file_test(
```

Comment:
Can we replace this test instead of removing it?
(i.e., what is the expectation now for the staging-service.substituted.yaml file to contain now that resolution has been moved to the resolver?)

Reply:
This PR removes the intermediate `X.substituted.yaml` entirely since substitutions are now handled by the resolver. So there is no file to test against.

Is there a use-case for having the substituted-but-unresolved intermediate available? I looked through the examples but couldn't find any actual usage. Though it is true, this would be a breaking change for any users who are actually using `X.substituted.yaml` for anything.

Reply:
hmm, I think it would not really make sense to depend on `X.substituted.yaml `. Lets clean up a bit by removing the substitutions in line 37 and the  `extratag: %{SUBSTITUTED_ARGUMENT}` in service.yaml. I would still like some test to verify the resolver is doing its job right wrt the changes you made.

Reply:
Sounds good, I'll work on adding a test.

Also, if we change our minds and want to have `X.subsituted.yaml` again, it would be possible to do that by writing a separate "substituter" that runs during build time. My feeling is that it's unnecessary complexity unless someone can articulate a use case, but I just wanted to point out that it can be done if needed.